src/main/java/ratpack/pac4j/RatpackPac4j.java

```diff
@@ -510,39 +499,43 @@ public static Operation logout(Context ctx) {
   private static void initiateAuthentication(Context ctx, Class<? extends Client> clientType) {
     Request request = ctx.getRequest();
     Clients clients = ctx.get(Clients.class);
-    Client<?, ?> client = clients.findClient(clientType);
+    Client client = clients.findClient(clientType).get();
 
     RatpackWebContext.from(ctx, false).then(webContext -> {
       webContext.getSessionStore().set(webContext, Pac4jSessionKeys.REQUESTED_URL.getName(), request.getUri());
       try {
-        client.redirect(webContext);
+        Optional<RedirectionAction> redirect = client.getRedirectionAction(webContext, webContext.getSessionStore());
+        if (redirect.isPresent() && redirect.get() instanceof WithLocationAction) {
+          ctx.redirect(((WithLocationAction) redirect.get()).getLocation());
+        } else {
+          ctx.error(new TechnicalException("Failed to redirect"));
+          webContext.sendResponse();
+        }
       } catch (Exception e) {
         if (e instanceof HttpAction) {
           webContext.sendResponse((HttpAction) e);
-          return;
         } else {
           ctx.error(new TechnicalException("Failed to redirect", e));
         }
       }
-
-      webContext.sendResponse();
```

Comment:
This change to the structure of how response sending is handled looks good to me.
I think that before, the `webContext.sendResponse()` would either be incorrect or redundant in the case of an error.